cmdstanpy/model.py

```diff
@@ -1341,22 +1355,37 @@ def _wrap_sampler_progress_hook(
         self, chain_id: int, total: int
     ) -> Optional[Callable[[str], None]]:
         """Sets up tqdm callback for CmdStan sampler console msgs."""
-        pbar: Any = tqdm(
-            total=total,
-            bar_format="{desc} |{bar}| {elapsed} {postfix[0][value]}",
-            postfix=[dict(value="Status")],
-            desc=f'chain {chain_id}',
-            colour='yellow',
-        )
+        try:
+            pbar: Any = tqdm(
+                total=total,
+                bar_format="{desc} |{bar}| {elapsed} {postfix[0][value]}",
+                postfix=[dict(value="Status")],
+                desc=f'chain {chain_id}',
+                colour='yellow',
+            )
+        # pylint: disable=broad-except
+        except Exception as e:
+            disable_progress(e)
+
+            # pylint: disable=unused-argument
+            def sampler_progress_hook(line: str) -> None:
+                return
+
+        else:
 
-        def sampler_progress_hook(line: str) -> None:
-            if line == "Done":
-                pbar.postfix[0]["value"] = 'Sampling completed'
-                pbar.close()
-            elif line.startswith("Iteration"):
-                if 'Sampling' in line:
-                    pbar.colour = 'blue'
-                pbar.update(1)
-                pbar.postfix[0]["value"] = line
+            def sampler_progress_hook(line: str) -> None:
+                try:
+                    if line == "Done":
+                        pbar.postfix[0]["value"] = 'Sampling completed'
+                        pbar.update(total - pbar.n)
+                        pbar.close()
+                    elif line.startswith("Iteration"):
+                        if 'Sampling' in line:
+                            pbar.colour = 'blue'
+                            pbar.update(1)
+                            pbar.postfix[0]["value"] = line
+                # pylint: disable=broad-except
+                except Exception as e:
```

Comment:
I don’t think it’s possible for this code to throw an exception due to user configuration, only due to a bug in our usage, so I don’t think trapping it is a good idea